.github/workflows/nix.yml

```diff
@@ -79,7 +79,8 @@ jobs:
       - name: try to authenticate to ripple Conan remote
         id: remote
         run: |
-          echo outcome=$(conan user --remote ripple ${{ secrets.CONAN_USERNAME }} --password ${{ secrets.CONAN_TOKEN }} && echo success || echo failure) | tee ${GITHUB_OUTPUT}
+          conan user --remote ripple ${{ secrets.CONAN_USERNAME }} --password ${{ secrets.CONAN_TOKEN }}
+          echo outcome=$([ $? -eq 0 ] && echo success || echo failure) | tee ${GITHUB_OUTPUT}
```

Comment:
The problem with creating your branch in the XRPLF repo is that it has those secrets configured, so the connection works. On a personal repo (or just one that doesn't have secrets), when the `conan user` command fails, the whole step fails. https://github.com/ximinez/rippled/actions/runs/6342827288/job/17229389904

There's an easy fix, though. The shell is running with `set -e`, so you can undo it by starting the `run:` with `set +e`.

Example from my Windows CI PR: https://github.com/XRPLF/rippled/actions/runs/6343277722/job/17230836193?pr=4596#step:9:28

Reply:
Thanks for pointing this out.
I have incorporated your suggestions in this commit (https://github.com/XRPLF/rippled/compare/ckeshava-patch-2...ckeshava:rippled:ckeshava-patch-2), but I don't have permissions to update this branch, because this is a branch on XRPLF's fork.

Do I need to create a new PR?

Reply:
I think we've fixed this in other places with `|| true`, which is the fix I prefer because it is highly specific. Only one command is permitted to fail.

Reply:
`|| true` will change the value of `$?`, won't it? That would defeat the whole point of this PR.

```
$ false ; [ $? -eq 0 ] && echo success || echo failure
failure

$ false || true ; [ $? -eq 0 ] && echo success || echo failure
success
```

Reply:
>  I don't have permissions to update this branch, because this is a branch on XRPLF's fork.

How did you create the branch in the first place?

Reply:
hmm, that's a good point. I was able to create a patch branch from the github UI. Perhaps the settings were modified later @intelliot  ?

Reply:
I wanted to share another option before I forget about it:
```
conan user --remote ripple ${{ secrets.CONAN_USERNAME }} --password ${{ secrets.CONAN_TOKEN }} \
    && echo outcome=success | tee ${GITHUB_OUTPUT} || echo outcome=failure | tee ${GITHUB_OUTPUT}
```
It's more verbose, but it doesn't require and `set +e` or `|| true` or any other workarounds. It might be a little more readable, too.

Reply:
Yet another option. A minimal change to the original. Just redirects output of `conan user` to stderr.

```
echo outcome=$(conan user --remote ripple ${{ secrets.CONAN_USERNAME }} --password ${{ secrets.CONAN_TOKEN }} >&2 && echo success || echo failure) | tee ${GITHUB_OUTPUT}
```

Reply:
@ximinez 
If the secrets are not configured, then `outcome` is set to `failure` in both the suggestions. Won't both of these commands stop the build process?

@thejohnfreeman Does redirecting to the stderr avoid terminating the bash script? If yes, then I'm inclined to incorporate your suggestion, for brevity.

Reply:
> @ximinez
> If the secrets are not configured, then `outcome` is set to `failure` in both the suggestions. Won't both of these commands stop the build process?

No. That's the reason it's written this way. Setting `outcome=failure` only affects the "upload dependencies to remote" and "recreate archive with dependencies" steps.

Reply:
Ah, the additional `tee ${GITHUB_OUTPUT}` in your suggestion protects the entire bash command from failure. I'm guessing the command will short-circuit to the next OR condition. I think I understand it now 👍

Reply:
> Yet another option. A minimal change to the original. Just redirects output of `conan user` to stderr.
> 
> ```
> echo outcome=$(conan user --remote ripple ${{ secrets.CONAN_USERNAME }} --password ${{ secrets.CONAN_TOKEN }} >&2 && echo success || echo failure) | tee ${GITHUB_OUTPUT}
> ```

This sounds like a winner.

Reply:
I don't have permissions to write into the branch :/
let me ask elliot and revert back to this PR

Reply:
> > I don't have permissions to update this branch, because this is a branch on XRPLF's fork.
> 
> How did you create the branch in the first place?

@ckeshava previously had write access, but this was changed to read in order to improve the security of this repo. In general, [contributions](https://github.com/XRPLF/rippled/blob/develop/CONTRIBUTING.md) should be developed in your personal fork.

@ckeshava : please open a new PR using a branch on your fork. Thanks!